src/server/stream_family.cc

```diff
@@ -2629,6 +2724,98 @@ void StreamFamily::XAck(CmdArgList args, ConnectionContext* cntx) {
   return;
 }
 
+void StreamFamily::XAutoClaim(CmdArgList args, ConnectionContext* cntx) {
+  ClaimOpts opts;
+  string_view key = ArgS(args, 0);
+  opts.group = ArgS(args, 1);
+  opts.consumer = ArgS(args, 2);
+  if (!absl::SimpleAtoi(ArgS(args, 3), &opts.min_idle_time)) {
+    return (*cntx)->SendError(kSyntaxErr);
+  }
+  if (opts.min_idle_time < 0) {
+    opts.min_idle_time = 0;
+  }
+
+  string_view start = ArgS(args, 4);
+  RangeId rs;
+  if (!ParseRangeId(start, &rs)) {
+    return;
+  }
+  if (rs.exclude && streamDecrID(&rs.parsed_id.val) != C_OK) {
+    return (*cntx)->SendError("invalid start ID for the interval", kSyntaxErrType);
+  }
+  opts.start = rs.parsed_id.val;
+
+  for (size_t i = 5; i < args.size(); ++i) {
+    ToUpper(&args[i]);
+    string_view arg = ArgS(args, i);
+    bool remaining_args = args.size() - i - 1 > 0;
+
+    if (remaining_args) {
+      if (arg == "COUNT") {
+        arg = ArgS(args, ++i);
+        if (!absl::SimpleAtoi(arg, &opts.count)) {
+          return (*cntx)->SendError(kInvalidIntErr);
+        }
+        if (opts.count <= 0) {
+          return (*cntx)->SendError("COUNT must be > 0");
+        }
+        continue;
+      }
+    }
+    if (arg == "JUSTID") {
+      opts.flags |= kClaimJustID;
+    } else {
+      return (*cntx)->SendError("Unknown argument given for XAUTOCLAIM command", kSyntaxErr);
+    }
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpAutoClaim(t->GetOpArgs(shard), key, opts);
+  };
+  OpResult<ClaimInfo> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
+
+  if (result.status() == OpStatus::KEY_NOTFOUND) {
+    cntx->SendError(NoGroupOrKey(key, opts.group));
+    return;
```

Comment:
Maybe do `return cntx->SendError();` like you do above for consistency?
(also below)

Reply:
you meant: (*cntx)->SendError ? Yes, I can change it to this.